src/seedu/addressbook/data/person/Address.java

```diff
@@ -36,21 +53,25 @@ public static boolean isValidAddress(String test) {
         return test.matches(ADDRESS_VALIDATION_REGEX);
     }
 
+    private String getAddressString() {
+        return block + COMMA + street + COMMA + unit + COMMA + postalCode;
+    }
+
     @Override
     public String toString() {
-        return value;
+        return getAddressString();
     }
 
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+                && getAddressString().equals(((Address) other).getAddressString())); // state check; // state check
```

Comment:
Do we have access to `other.getAddressString()`? It is private. Consider using `toString()` instead.